base/core/src/main/java/org/openscience/cdk/ringsearch/RegularCyclicVertexSearch.java

```diff
@@ -54,7 +54,7 @@
     private long[]         state;
 
     /** Vertex colors - which component does each vertex belong. */
-    private volatile int[] colors;
+    private int[] colors;
```

Comment:
Because the concurrency is guaranteed working or so?

Reply:
Actually more that it would be odd for multiple threads to call this method and so it doesn't need to be volatile.

Reply:
I wrote the code... it was trying to do lazy initialisation. But really I wasn't thinking and lazy initalization doesn't need to be thread safe on this object. It does on the builders hence why I switched those to an enum. The reason it doesn't need to be thread safe is because we expect each thread to hold it's own copy. Even if there is a race condition and this array gets initalized twice, it's not an issue as not expensive.